src/testE2E/codecatalyst/client.test.ts

```diff
@@ -147,6 +160,37 @@ describe('Test how this codebase uses the CodeCatalyst API', function () {
             // and then continue this test from that point.
         })
 
+        it('prompts to install the ssh extension if not available', async function () {
+            if (isExtensionInstalled(VSCODE_EXTENSION_ID.remotessh)) {
+                this.skip()
+            }
+
+            await assert.rejects(
+                prepareDevEnvConnection(client, { ...defaultDevEnv }),
+                /Remote SSH extension not installed/
+            )
+            const notification = await getTestWindow().waitForMessage(
+                /Connecting to Dev Environment requires the Remote SSH extension/
+            )
+
+            // Normally selecting `Install...` would open up the extensions tab
+            // But for this test we actually do want to install the extension programtically
```

Comment:
is this because the tests need the extension? or why don't we just test whatever the actual behavior is?

Reply:
> is this because the tests need the extension? or why don't we just test whatever the actual behavior is?

Yes, the subsequent tests fail without the extension. We can install the extension prior to running the tests but that means we can't easily test the behavior without the extension. Which I think is import when considering the new user experience. 

We are testing (most) of the actual behavior but because we can't manipulate the extensions tab we instead simulate what a user might do e.g. install the extension.